rust/cadeau/src/xmf/vpx/mod.rs

```diff
@@ -0,0 +1,238 @@
+use core::fmt;
+
+use xmf_sys::{
+    XmfVpxCodecType, XmfVpxDecoderError, XmfVpxEncoderError, XmfVpxFrame, XmfVpxFrame_Destroy, XmfVpxFrame_GetBuffer,
+    XmfVpxFrame_GetDuration, XmfVpxFrame_GetFlags, XmfVpxFrame_GetHeight, XmfVpxFrame_GetPartitionId,
+    XmfVpxFrame_GetPts, XmfVpxFrame_GetSize, XmfVpxFrame_GetSpatialLayerEncoded, XmfVpxFrame_GetWidth, XmfVpxImage,
+    XmfVpxImage_Destroy, XmfVpxPacket, XmfVpxPacketKind, XmfVpxPacket_Destroy, XmfVpxPacket_GetFrame,
+    XmfVpxPacket_GetKind, XmfVpxPacket_IsEmpty,
+};
+
+pub mod decoder;
+pub mod encoder;
+
+#[derive(Debug, Clone, Copy)]
+pub enum VpxCodec {
+    VP8,
+    VP9,
+}
+
+pub struct VpxImage {
+    pub(crate) ptr: *mut XmfVpxImage,
+}
+
+impl Drop for VpxImage {
+    fn drop(&mut self) {
+        // Safety: it is safe to call, the pointer is owned by the VpxImage itself.
+        unsafe {
+            XmfVpxImage_Destroy(self.ptr);
+        }
+    }
+}
```

Comment:
question: Did you forget to wrap `XmfVpxImage_GetData`?

Reply:
My mistake, it should not be exported